internal/services/paloalto/next_generation_firewall_vhub_local_rulestack_resource.go

```diff
@@ -0,0 +1,301 @@
+package paloalto
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/paloaltonetworks/2022-08-29/firewalls"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/paloaltonetworks/2022-08-29/localrulestacks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/paloalto/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/paloalto/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type NextGenerationFirewallVHubLocalRuleStackResource struct{}
+
+type NextGenerationFirewallVHubLocalRuleStackModel struct {
+	Name              string `tfschema:"name"`
+	ResourceGroupName string `tfschema:"resource_group_name"`
+	RuleStackId       string `tfschema:"rule_stack_id"`
+
+	NetworkProfile []schema.NetworkProfileVHub `tfschema:"network_profile"`
+	DNSSettings    []schema.DNSSettings        `tfschema:"dns_settings"`
+	FrontEnd       []schema.DestinationNAT     `tfschema:"destination_nat"`
+
+	// Computed
+	PlanData []schema.Plan `tfschema:"plan"`
+	PanEtag  string        `tfschema:"pan_etag"`
+
+	Tags map[string]interface{} `tfschema:"tags"`
+}
+
+var _ sdk.ResourceWithUpdate = NextGenerationFirewallVHubLocalRuleStackResource{}
+
+func (r NextGenerationFirewallVHubLocalRuleStackResource) ModelObject() interface{} {
+	return &NextGenerationFirewallVHubLocalRuleStackModel{}
+}
+
+func (r NextGenerationFirewallVHubLocalRuleStackResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return firewalls.ValidateFirewallID
+}
+
+func (r NextGenerationFirewallVHubLocalRuleStackResource) ResourceType() string {
+	return "azurerm_palo_alto_next_generation_firewall_vhub_local_rulestack"
```

Comment:
presumably this should be:

```suggestion
	return "azurerm_palo_alto_next_generation_firewall_virtual_hub_local_rulestack"
```